dotCMS/src/main/java/com/dotmarketing/portlets/structure/factories/FieldFactory.java

```diff
@@ -97,22 +136,28 @@ public static Field getFieldByStructure(String structureInode, String fieldName)
         return new Field();
     }
 
-
 	//### CREATE AND UPDATE ###
-	public static Field saveField(Field oldField) throws DotHibernateException
-	{
-	    
-        if (oldField.getFieldType().equals("host or folder") || oldField.getFieldType().equals("line_divider") || oldField.getFieldType().equals("tab_divider")
-                || oldField.getFieldType().equals("categories_tab")
-                || oldField.getFieldType().equals("permissions_tab") 
-                || oldField.getFieldType().equals( "relationships_tab")
-                || oldField.getFieldType().equals("category")
-                || oldField.getFieldType().equals("tag")
-             ) {
-
-            oldField.setFieldContentlet(DataTypes.SYSTEM.toString());
-            
-        }
+	/**
+	 * Saves a field in a Content Type.
+	 * 
+	 * @param oldField
+	 *            - The legacy Field object.
+	 * @return The Field object that was saved.
+	 * @throws DotHibernateException
+	 *             An error occurred when saving the field.
+	 */
+	public static Field saveField(Field oldField) throws DotHibernateException {
+		if (Field.FieldType.HOST_OR_FOLDER.toString().equals(oldField.getFieldType())
```

Comment:
It seems that in the new FieldFactory this is already checked https://github.com/dotCMS/core/blob/master-4.1/dotCMS/src/main/java/com/dotcms/contenttype/business/FieldFactoryImpl.java#L230